### PR TITLE
LIME-259: * Rewritting selectors to fix warnings

### LINKED ISSUE #259: Fix warnings
Fix warnings

Image

### PR DESCRIPTION
**Pull request by DavidCharif**
Local navigation through routes without warnings
image

### CODE REVIEW

File: frontend/src/bundles/auth/pages/auth.tsx [hunk 1]
```diff
@@ -37,15 +37,10 @@ const Auth: React.FC = () => {
...
-        dataStatus: auth.dataStatus,
-        user: auth.user,
-    }));
-
-    const { dataStatus: resetPasswordStatus } = useAppSelector(
-        ({ passwordReset }) => ({
-            dataStatus: passwordReset.dataStatus,
-        }),
+    const dataStatus = useAppSelector(({ auth }) => auth.dataStatus);
+    const user = useAppSelector(({ auth }) => auth.user);
```
Comment (dimapopovych) [inline frontend/src/bundles/auth/pages/auth.tsx]: you don't need to create separate selectors to get data from one slice, could select whole slice and use destructuring

**Review by dimapopovych — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/auth/pages/auth.tsx (hunk 1 above)
Comment (nikita-remeslov) [inline frontend/src/bundles/auth/pages/auth.tsx]:
```suggestion
    const { dataStatus, user } = useAppSelector(({ auth }) => auth);
```

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/auth/pages/auth.tsx (hunk 1 above)
Comment (DavidCharif) [reply frontend/src/bundles/auth/pages/auth.tsx]: image

File: frontend/src/bundles/common/components/protected-route/protected-route.tsx [hunk 2]
```diff
@@ -8,10 +8,9 @@ type ProtectedRouteProperties = {
 };
 
 const ProtectedRoute: React.FC<ProtectedRouteProperties> = ({ children }) => {
-    const { isRefreshing, userAuthenticated } = useAppSelector(({ auth }) => ({
-        isRefreshing: auth.isRefreshing,
-        userAuthenticated: auth.user,
-    }));
+    const { isRefreshing, user: userAuthenticated } = useAppSelector(
```
Comment (dimapopovych) [inline frontend/src/bundles/common/components/protected-route/protected-route.tsx]: I don't think that renaming here is necessary

File: frontend/src/bundles/goals/pages/goals.tsx [hunk 3]
```diff
@@ -48,18 +48,14 @@ const ZERO_VALUE = 0;
...
     const dispatch = useAppDispatch();
 
-    const { dataStatus: dataStatusGoals, goals } = useAppSelector(
-        ({ goals }) => ({
-            dataStatus: goals.dataStatus,
-            goals: goals.goals,
-        }),
-    );
+    const dataStatusGoals = useAppSelector(({ goals }) => goals.dataStatus);
+    const goals = useAppSelector(({ goals }) => goals.goals);
```
Comment (dimapopovych) [inline frontend/src/bundles/goals/pages/goals.tsx]: multiple selectors is unnecessary

File: frontend/src/bundles/goals/pages/goals.tsx [hunk 4]
```diff
@@ -48,18 +48,14 @@ const ZERO_VALUE = 0;
...
-    const { dataStatus: dataStatusAchievements, achievements } = useAppSelector(
-        ({ achievements }) => ({
-            dataStatus: achievements.dataStatus,
-            achievements: achievements.achievements,
-        }),
+    const dataStatusAchievements = useAppSelector(
+        ({ achievements }) => achievements.dataStatus,
+    );
+    const achievements = useAppSelector(
+        ({ achievements }) => achievements.achievements,
```
Comment (dimapopovych) [inline frontend/src/bundles/goals/pages/goals.tsx]: multiple selectors is unnecessary

**Review by dimapopovych — CHANGES_REQUESTED** [review]

**Comment by nikita-remeslov** [comment]
Conflicts

**Review by stpalamar — DISMISSED** [review]

**Review by dimapopovych — DISMISSED** [review]

**Review by ArtemLymarenko — DISMISSED** [review]

File: frontend/src/bundles/common/components/theme-switcher/theme-switcher.tsx [hunk 5]
```diff
@@ -21,7 +21,8 @@ function ThemeSwitcher({
     className = defaultClassName,
 }: Properties): JSX.Element {
     const dispatch = useAppDispatch();
-    const { theme } = useAppSelector((state) => state.theme);
+    const theme = useAppSelector(({ theme }) => theme.theme);
```
Comment (valery-chumak) [inline frontend/src/bundles/common/components/theme-switcher/theme-switcher.tsx]: const { theme } = useAppSelector(({ theme }) => theme);

File: frontend/src/bundles/landing/pages/landing.tsx [hunk 6]
```diff
@@ -37,7 +37,7 @@ import {
 import { FEATURES, TESTIMONIALS } from '../constants/constants.js';
 
 const Landing = (): JSX.Element => {
-    const { theme } = useAppSelector(({ theme }) => theme);
+    const theme = useAppSelector(({ theme }) => theme.theme);
```
Comment (valery-chumak) [inline frontend/src/bundles/landing/pages/landing.tsx]: const { theme } = useAppSelector(({ theme }) => theme);

File: frontend/src/bundles/profile/pages/connections-page/components/connection-option.tsx [hunk 7]
```diff
@@ -24,10 +24,9 @@ const ConnectionOption = ({
 }: TConnectionOption): JSX.Element => {
     const dispatch = useAppDispatch();
 
-    const { connections } = useAppSelector(({ connections }) => ({
-        dataStatus: connections.dataStatus,
-        connections: connections.connections,
-    }));
+    const connections = useAppSelector(
```
Comment (valery-chumak) [inline frontend/src/bundles/profile/pages/connections-page/components/connection-option.tsx]: const { connections } = useAppSelector(({ connections }) => connections);

File: frontend/src/bundles/workouts/pages/workout-item/workout-items.tsx [hunk 8]
```diff
@@ -12,9 +12,7 @@ import { OAuthProvider } from '~/bundles/workouts/enums/enums.js';
 const WorkoutItem = (): JSX.Element => {
     const { id } = useParams();
 
-    const { workouts } = useAppSelector(({ workouts }) => ({
-        workouts: workouts.workouts,
-    }));
+    const workouts = useAppSelector(({ workouts }) => workouts.workouts);
```
Comment (valery-chumak) [inline frontend/src/bundles/workouts/pages/workout-item/workout-items.tsx]: const { workouts } = useAppSelector(({ workouts }) => workouts );

Comment (stpalamar) [inline frontend/src/bundles/workouts/pages/workout-item/workout-items.tsx]:
```suggestion
    const { workouts } = useAppSelector(({ workouts }) => workouts);
```

File: frontend/src/bundles/subscription/pages/subscription-checkout/subscription-checkout.tsx [hunk 9]
```diff
@@ -10,7 +10,7 @@ import { config } from '~/framework/config/config.js';
 const stripe = loadStripe(config.ENV.STRIPE.PUBLIC_KEY);
 
 const SubscriptionCheckout = (): JSX.Element => {
-    const { theme } = useAppSelector((state) => state.theme);
+    const theme = useAppSelector(({ theme }) => theme.theme);
```
Comment (valery-chumak) [inline frontend/src/bundles/subscription/pages/subscription-checkout/subscription-checkout.tsx]: const { theme } = useAppSelector(({ theme }) => theme);

**Review by valery-chumak — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/workouts/pages/workout-item/workout-items.tsx (hunk 8 above)
Comment (stpalamar) [reply frontend/src/bundles/workouts/pages/workout-item/workout-items.tsx]: no a big deal but it better follow the same code style)

Comment (stpalamar) [reply frontend/src/bundles/workouts/pages/workout-item/workout-items.tsx]: like Valeriia already mentioned above)

**Review by valery-chumak — APPROVED** [review]

**Review by ArtemLymarenko — APPROVED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**